contracts/jobs/JobsManager.sol

```diff
@@ -68,96 +154,329 @@ contract JobsManager is IJobsManager, Verifiable, Controllable {
+        // Transcoder must have missed verification for the segment
+        require(!claim.segmentVerifications[_segmentNumber]);
+
+        refundBroadcaster(_jobId, _claimId);
+
+        // Slash transcoder and provide finder params
+        bondingManager().slashTranscoder(job.transcoderAddress, msg.sender, missedVerificationSlashAmount, finderFee);
+
+        // Set claim as slashed
+        claim.status = ClaimStatus.Slashed;
+
+        return true;
     }
 
     /*
-     * @dev Provide proof of transcoding a segment
-     * TODO: This might have to be payable when we implement the transcoding verification process using a solution such as Oraclize or Truebit
+     * @dev Distribute fees for a particular claim
      * @param _jobId Job identifier
-     * @param _segmentSequenceNumber Segment sequence number in stream
-     * @param _dataHash Segment data hash. Used to retrieve segment data from Swarm
-     * @param _transcodedDataHash Transcoded segment data hash. Used to retrieve transcoded segment data from Swarm
-     * @param _broadcasterSig Broadcaster's signature over segment
-     * @param _proof Merkle proof for the signed transcode claim
+     * @param _claimId Claim identifier
      */
-    function verify(uint256 _jobId, uint256 _segmentSequenceNumber, string _dataHash, string _transcodedDataHash, bytes _broadcasterSig, bytes _proof) payable external returns (bool) {
-        if (!jobs.validateTranscodeClaim(_jobId, _segmentSequenceNumber, _dataHash, _transcodedDataHash, _broadcasterSig, _proof, verificationRate)) throw;
+    function distributeFees(uint256 _jobId, uint256 _claimId)
+        public
+        whenSystemNotPaused
+        jobExists(_jobId)
+        returns (bool)
+    {
+        Job storage job = jobs[_jobId];
+        Claim storage claim = job.claims[_claimId];
 
-        // TODO: use a real verification code hash
-        string memory verificationCodeHash = "123";
-        // Invoke transcoding verification. This is async and will result in a callback to receiveVerification() which is implemented by this contract
-        verifier.verify(_jobId, _segmentSequenceNumber, verificationCodeHash, _dataHash, _transcodedDataHash, this);
+        // Sender must be elected transcoder for job
+        require(job.transcoderAddress == msg.sender);
+        // Claim must not be complete
+        require(claim.status == ClaimStatus.Pending);
+        // Slashing period must be over for claim
+        require(claim.endSlashingBlock < block.number);
+
+        uint256 fees = claim.segmentRange[1].sub(claim.segmentRange[0]).add(1).mul(job.pricePerSegment);
+        // Deduct fees from escrow
+        job.escrow = job.escrow.sub(fees);
+        // Add fees to transcoder's fee pool
+        bondingManager().updateTranscoderFeePool(msg.sender, fees, claim.claimBlock, claim.transcoderTotalStake);
+        // Send fees to bonding manager
+        token.transfer(address(bondingManager()), fees);
```

Comment:
This was missing before. Now this function actually sends tokens deposited to the JobsManager to the BondingManager so the BondingManager can send tokens to delegators and transcoders when they withdraw